internal/verify/enum.go

```diff
@@ -0,0 +1,41 @@
+package verify
+
+import (
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+type EnumValues[T ~string] interface {
+	~string
+	Values() []T
+}
+
+// ValidateEnum creates a schema validation function for the provided enum type
+func ValidateEnum[T EnumValues[T]]() schema.SchemaValidateDiagFunc {
+	values := filterUnknownValues(getValues[T]())
+	return validation.ToDiagFunc(validation.StringInSlice(values, false))
```

Comment:
`ignoreCase` now default to `false` where it was previously `true`

Reply:
`ignoreCase` has always been defaulted to false, the only case I found where its set to true is on [pat_rule's protocol field](https://github.com/scaleway/terraform-provider-scaleway/blob/master/internal/services/vpcgw/pat_rule.go#L72)